### ISSUE TITLE
Weird behavior of affine transformations

### ISSUE THREAD

**Issue by clementpoiret** [neurospin/pynet#18]
Hi, and sorry for opening many issues in a short period of time :)

I'm facing a strange issue w/ affine transformations (rotation).
I'm working with a segmentation dataset, therefore, I need to apply the exact same transformation to my MRI and its manual labels.
Thus, I wrapped `affine()` into a `RandomAffine` class, appling `map_coordinate(...)` to both objects... And it's changing the unique values of my ground truth from [0,1,2] to [-1,0,1,2,3] even if using rotations only.

Shouldn't an affine transformation just move points in space w/o changing their values?

Here is a little code snippet to replicate the issue: <https://colab.research.google.com/drive/1alUMLINz9n9mpmCwMQZPkr7-phbUlYSv?usp=sharing>

Thanks!

**Comment by AGrigis** [comment]
Hi Clement,

pynet is already doing that for you.
When you use the Transformer class to define your set of  transformations, you can control the interpolation via the 'output_label' parameter and also the type of data to apply the transform to via the 'apply_to' parameter.
Then the DataManager through the created ArrayDatasets will set automatically the 'seed' and 'dtype'  parameters to ensure that an input array and for instance his associated segmentation array will be transformed the same way.

For the other issue, you didn't specify the appropriate interpolation. You need to set a nearest neighbour interpolation on labels via 'order=0'. I set a simple example below:

import numpy as np
import torch
from pynet.augmentation import affine

x = torch.randn(32, 32, 32)
y = torch.randint_like(x, 3).int().numpy()
y_new = affine(y, rotation=0, translation=5, zoom=0, order=0)
np.unique(y_new)
      array([0, 1, 2], dtype=int32)
np.unique(y)
      array([0, 1, 2], dtype=int32)

Best.

**Comment by clementpoiret** [comment]
Hi Antoine,

Thanks a lot for your answer! I'll take a look at the Transformer class :+1: 
For the interpolation, I didn't thought it'd have an effect for a linear translation too, but `order=0` for labels solves the issue.

Thanks again!
